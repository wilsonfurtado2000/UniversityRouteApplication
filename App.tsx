import React from 'react';
import {StyleSheet} from 'react-native';
import {connect} from 'react-redux';
import AppNavigator from './navigation/AppNavigator';
import {NavigationContainer} from '@react-navigation/native';
import ScreenTime from './screenTime/ScreenTime';

function App({isLoggedIn}) {
  return (
    <NavigationContainer>
      <ScreenTime />
      <AppNavigator isLoggedIn={isLoggedIn} />
    </NavigationContainer>
  );
}

const mapStateToProps = state => ({
  isLoggedIn: !!state.user.userEmail,
});

export default connect(mapStateToProps)(App);
