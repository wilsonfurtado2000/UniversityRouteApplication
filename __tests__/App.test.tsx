import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import App from '../App';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  NavigationContainer: ({children}) => <div>{children}</div>,
}));
jest.mock('../navigation/AppNavigator', () => props => (
  <div {...props}>AppNavigator</div>
));
jest.mock('../screenTime/ScreenTime', () => 'ScreenTime');

const reducer = (state = {user: {userEmail: null}}, action) => state;
const store = createStore(reducer);

describe('App', () => {
  const createTestStore = userEmail => createStore(() => ({user: {userEmail}}));

  it('renders as logged in', () => {
    const loggedInStore = createTestStore('user@example.com');
    const tree = renderer
      .create(
        <Provider store={loggedInStore}>
          <App />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders as logged out', () => {
    const loggedOutStore = createTestStore(null);
    const tree = renderer
      .create(
        <Provider store={loggedOutStore}>
          <App />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes isLoggedIn to AppNavigator', () => {
    const store = createTestStore('user@example.com');
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    const appNavigator = component.root.findByProps({children: 'AppNavigator'});
    expect(appNavigator.props.isLoggedIn).toBe(true);
  });
});
