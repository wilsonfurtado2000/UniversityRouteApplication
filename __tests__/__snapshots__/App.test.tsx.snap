
exports[`App renders as logged in 1`] = `
<div>
  <ScreenTime />
  <div
    isLoggedIn={true}
  >
    AppNavigator
  </div>
</div>
`;

exports[`App renders as logged out 1`] = `
<div>
  <ScreenTime />
  <div
    isLoggedIn={false}
  >
    AppNavigator
  </div>
</div>
`;

exports[`App renders correctly 1`] = `
<div>
  <ScreenTime />
  <AppNavigator
    isLoggedIn={false}
  />
</div>
`;

exports[`App with different authentication states renders as logged in 1`] = `
<div>
  <ScreenTime />
  <AppNavigator
    isLoggedIn={true}
  />
</div>
`;

exports[`App with different authentication states renders as logged out 1`] = `
<div>
  <ScreenTime />
  <AppNavigator
    isLoggedIn={false}
  />
</div>
`;
